contrib/epee/src/net_ssl.cpp

```diff
@@ -522,30 +522,101 @@ bool ssl_options_t::handshake(
       return true;
     });
   }
+}
+
+bool ssl_options_t::handshake(
+  boost::asio::ssl::stream<boost::asio::ip::tcp::socket> &socket,
+  boost::asio::ssl::stream_base::handshake_type type,
+  boost::asio::const_buffer buffer,
+  const std::string& host,
+  std::chrono::milliseconds timeout) const
+{
+  configure(socket, type, host);
```

Comment:
There's no reason to include these SSL changes with the connection re-write. Include these changes in a new PR with an explanation as to why its being changed.

And if the issue being fixed is the racy `ec = ...`, this can be fixed with less changes. All that was needed is an `std::atomic_flag` to signal the change (a safe "happens before/after synchronization), these other changes are overkill.